test/test_automl/test_automl.py

```diff
@@ -334,6 +329,52 @@ def test_automl_outputs(backend, dask_client):
         start_time = float(fh.read())
     assert time.time() - start_time >= 10, print_debug_information(auto)
 
+    # Then check that the logger matches the run expectation
+    logfile = glob.glob(os.path.join(
+           auto._backend.temporary_directory, 'AutoML*.log'))[0]
+    parser = AutoMLLogParser(logfile)
+
+    # The number of ensemble trajectories properly in log file
+    success_ensemble_iters_auto = len(auto.ensemble_performance_history)
+    success_ensemble_iters_log = parser.count_ensembler_success_pynisher_calls()
+    assert success_ensemble_iters_auto == success_ensemble_iters_log, "{} != {}".format(
+        auto.ensemble_performance_history,
+        print_debug_information(auto),
+    )
+
+    # We also care that no iteration got lost
+    # This is important because it counts for pynisher calls
+    # and whether a pynisher call actually called the ensemble
+    total_ensemble_iterations = parser.count_ensembler_iterations()
+    assert len(total_ensemble_iterations) > 1  # At least 1 iteration
+    assert range(1, max(total_ensemble_iterations) + 1), total_ensemble_iterations
+
+    # The number of completed tae calls must be in the log file
+    # Running configurations cannot be check as they might not reached
+    # a point where pynisher is called before budget exhaustion
+    total_completed_runs_log = parser.count_tae_pynisher_calls() - 1  # Dummy not in run history
+    total_completed_runs_auto = 0
+    for run_value in auto.runhistory_.data.values():
+        # Make sure that running jobs are properly tracked. Killed runs do not always
+        # print the return value to the log file (yet the run history has this information)
+        if run_value.status == StatusType.SUCCESS:
+            # Success is not sufficient to guarantee a return message in the log file
```

Comment:
Just checking, if the run doesn't print `r'pynisher]\s+return value:\s+'` it should print either or ` "Your function call closed the pipe prematurely -> Subprocess probably got an uncatchable signal."` or `"Something else went wrong, sorry."` in the controlling process, so we could actually check that, right?

Reply:
If a failure happens [here](https://github.com/automl/pynisher/blob/beb76e4394d364efb4847809a2f06ec5e1f18d54/pynisher/limit_function_call.py#L154) we will get a double count message, because while sending the pipe message it could break. 

However, we can make this deterministic and add a message at the end of the pynisher function call (inside pynisher)? For instance [here](https://github.com/automl/pynisher/blob/beb76e4394d364efb4847809a2f06ec5e1f18d54/pynisher/limit_function_call.py#L280)

Reply:
But aren't the two error messages I mentioned exactly the error messages showing up in case of a failure to write to the pipe?